tests/test_dnsmasq_server.cpp

```diff
@@ -67,7 +65,6 @@ struct DNSMasqServer : public mpt::TestWithMockedBinPath
 
     mpt::TempDir data_dir;
     std::shared_ptr<CapturingLogger> logger = std::make_shared<CapturingLogger>();
-    mpt::StubProcessFactory process_factory;
     const QString bridge_name{"dummy-bridge"};
```

Comment:
Isn't this missing a `StubProcessFactory::Inject()`, here and in libvirt tests? Otherwise this now creates actual processes, while that was not the case before.

Reply:
This test actually does create actual processes, its test harness `mpt::TestWithMockedBinPath`  messes with the $PATH to launch a test binary.

I thought this change was most minimal to keep the test working - and restores it to be more like the original design. Likely I'll refactor this test to use MockProcessFactory later.

Reply:
Alright